src/bundle/Resources/public/js/scripts/fieldType/base/base-rich-text.js

```diff
@@ -170,8 +172,14 @@
 
         clearAnchor(element) {
             const icon = element.querySelector('.ez-icon--anchor');
-
-            if (icon) {
+            const elementPreviousSibling = element.previousSibling;
+            const isTableWithAnchor =
+                element.tagName.toLowerCase() === TABLE_TAG_NAME &&
+                elementPreviousSibling.tagName.toLowerCase() === SVG_TAG_NAME;
```

Comment:
`element.previousSibling` can be `null`, which would cause the script to crash due to method call on an undefined.

Reply:
`clearAnchor` is called within the following context:
```
documentFragment.querySelectorAll('.ez-has-anchor').forEach(this.clearAnchor);
```
If there are no elements having this class, there will be no method call, so no potential null for the `element` variable I suppose.

Reply:
I'm not confident that it's an assumption that we can make (i.e. that the context won't change or the HTML elements are not changed). I'd rather have an additional check that we don't get `null` by chance.

And it's actually fairly easy to do by using the [Optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). It has the same compatibility as object destructuring, so we should be safe to use it?

```suggestion
            const elementPreviousSibling = element.previousSibling;
            const isTableWithAnchor =
                element.tagName.toLowerCase() === TABLE_TAG_NAME &&
                elementPreviousSibling?.tagName.toLowerCase() === SVG_TAG_NAME;
```

Reply:
This assumption here is ok as we have the BC and the HTML cannot really change especially in the LTS version, so this is nice to have but not mandatory in that case.
One thing, I would not like to introduce optional chaining in the old LTS code.

Reply:
We might introduce that in the PR targetting v3: https://github.com/ezsystems/ezplatform-richtext/pull/182. WDYT @dew326?

Reply:
You can also just check the object:

```suggestion
            const elementPreviousSibling = element.previousSibling;
            const isTableWithAnchor =
                element.tagName.toLowerCase() === TABLE_TAG_NAME &&
                elementPreviousSibling && 
                elementPreviousSibling.tagName.toLowerCase() === SVG_TAG_NAME;
```

Reply:
@konradoboza yep, optional chaining sounds better for 3.x